org.envirocar.app/src/org/envirocar/app/views/tracklist/TrackListLocalCardFragment.java

```diff
@@ -157,6 +159,13 @@ public void onDeleteTrackClicked(Track track) {
             @Override
             public void onUploadTrackClicked(Track track) {
                 LOG.info(String.format("onUploadTrackClicked(%s)", track.getTrackID()));
+
+                // check if device is connected to internet before uploading the track
+                if (!contextInternetAccessProvider.isConnected() && getView() != null) {
```

Comment:
Will fail always after the first request because the application context you are using for initializing the ContextInternetAccessProvider becomes `null`.

Reply:
sorry for the inconvenience but sir I did test it many times and it always worked without any crashes. should I share a recording?
I can maybe create an object there in the if condition?  
```
new ContextInternetAccessProvider(getContext()).isConnected()
```

Reply:
sorry for the inconvenience sir @SebaDro in ``ContextInternetAccessProvider`` class there was this block 
```
        if (context == null)
            return false;
```
so I was never able to notice if anything was going wrong, but thanks for telling me. I will use this new approach and push right now, I hope you can look into this soon.
```
new ContextInternetAccessProvider(getContext()).isConnected()
```

Reply:
Yes, that's the reason why I suggested to extensively test your implementations. If you did so, you would have noticed it ;-)